homeassistant/components/tautulli/config_flow.py

```diff
@@ -0,0 +1,142 @@
+    ) -> FlowResult:
+        """Handle a flow initiated by the user."""
+        if self._async_current_entries():
+            return self.async_abort(reason="single_instance_allowed")
+        errors = {}
+        if user_input is not None:
+            if (error := await self.validate_input(user_input)) is None:
+                return self.async_create_entry(
+                    title=DEFAULT_NAME,
+                    data=user_input,
+                )
+            errors["base"] = error
+
+        user_input = user_input or {}
+        data_schema = {
+            vol.Required(CONF_API_KEY, default=user_input.get(CONF_API_KEY, "")): str,
+            vol.Required(CONF_URL, default=user_input.get(CONF_URL, "")): str,
+            vol.Optional(
+                CONF_VERIFY_SSL, default=user_input.get(CONF_VERIFY_SSL, True)
+            ): bool,
+        }
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(data_schema),
+            errors=errors or {},
+        )
+
+    async def async_step_reauth(self, config: dict[str, Any]) -> FlowResult:
+        """Handle a reauthorization flow request."""
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(
+        self, user_input: dict[str, str] | None = None
+    ) -> FlowResult:
+        """Confirm reauth dialog."""
+        errors = {}
+        if user_input is not None and (
+            entry := self.hass.config_entries.async_get_entry(self.context["entry_id"])
+        ):
+            _input = {**entry.data, CONF_API_KEY: user_input[CONF_API_KEY]}
+            if (error := await self.validate_input(_input)) is None:
+                self.hass.config_entries.async_update_entry(entry, data=_input)
+                await self.hass.config_entries.async_reload(entry.entry_id)
+                return self.async_abort(reason="reauth_successful")
+            errors["base"] = error
+        return self.async_show_form(
+            step_id="reauth_confirm",
+            data_schema=vol.Schema({vol.Required(CONF_API_KEY): str}),
+            errors=errors,
+        )
+
+    async def async_step_import(self, config: dict[str, Any]) -> FlowResult:
+        """Import a config entry from configuration.yaml."""
+        _LOGGER.warning(
+            "Configuration of the Tautulli platform in YAML is deprecated and will be "
+            "removed in Home Assistant 2022.6; Your existing configuration for host %s"
+            "has been imported into the UI automatically and can be safely removed "
+            "from your configuration.yaml file",
+            config[CONF_HOST],
+        )
+        if self._async_current_entries():
+            return self.async_abort(reason="single_instance_allowed")
+        host_configuration = PyTautulliHostConfiguration(
+            config[CONF_API_KEY],
+            ipaddress=config[CONF_HOST],
+            port=config.get(CONF_PORT, DEFAULT_PORT),
```

Comment:
The port should be an integer. The old config schema and `DEFAULT_PORT` returns a string. That doesn't look right. We should adjust that to stay in line with the library model typing.